iothub/device/src/Pipeline/ExceptionRemappingHandler.cs

```diff
@@ -128,56 +130,50 @@ private static bool IsNetwork(Exception singleException)
             return s_networkExceptions.Any(baseExceptionType => baseExceptionType.IsInstanceOfType(singleException));
         }
 
-        private Task ExecuteWithErrorHandlingAsync(Func<Task> asyncOperation)
+        private Task ExecuteWithExceptionRemappingAsync(Func<Task> asyncOperation)
         {
-            return ExecuteWithErrorHandlingAsync(async () =>
+            return RunWithExceptionRemappingAsync(async () =>
             {
                 await asyncOperation().ConfigureAwait(false);
                 return false;
             });
         }
 
-        private async Task<T> ExecuteWithErrorHandlingAsync<T>(Func<Task<T>> asyncOperation)
+        private async Task<T> RunWithExceptionRemappingAsync<T>(Func<Task<T>> asyncOperation)
         {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"{nameof(ExceptionRemappingHandler)}.{nameof(ExecuteWithExceptionRemappingAsync)}");
+
             try
+            {
+                return await asyncOperation().ConfigureAwait(false);
+            }
+            catch (Exception ex) when (ex is not IotHubClientException && !Fx.IsFatal(ex))
```

Comment:
Adding this exception (pun intended) to the catch makes sense to me and eliminates the small block on former lines 162-166. What do you think?

If the job here is to not allow most exceptions that aren't IotHubClientException, why would we catch when it isn't?